src/js/app.js

```diff
@@ -1,98 +1,82 @@
-        <input type="checkbox" ${task.isCompleted ? 'checked' : ''} />
-        <p>${task.description}</p>
-      </div>
-      <i class="fa-solid fa-ellipsis-vertical draggable options-icon"></i>
-    </article>
-  </li>
-  `;
-  tasksListEl.insertAdjacentHTML('beforeend', content);
+    localStorage.setItem('todo', JSON.stringify(toDoList));
+    tasksListEl.innerHTML = '';
+    displayToDoList();
+  });
 };
 
-const handleTaskInputsClick = (id) => {
+const handleTaskBtnsClick = (id, displayToDoList) => {
   const taskEl = document.getElementById(`${id}`);
 
-  const taskCheckBox = taskEl.querySelector('input');
-  const taskdescription = taskEl.querySelector('p');
+  const taskCheckBox = taskEl.querySelector('.task_checkbox');
+  const taskDeleteIcon = taskEl.querySelector('.delete-option-icon');
+  const taskEditIcon = taskEl.querySelector('.edit-option-icon');
+  const taskEditInput = taskEl.querySelector('.task_description');
+  const taskDescription = taskEl.querySelector('p');
 
-  taskCheckBox.addEventListener('click', () => {
-    if (taskCheckBox.checked) {
-      taskdescription.style.textDecorationLine = 'line-through';
-    } else {
-      taskdescription.style.textDecorationLine = 'none';
-    }
+  taskCheckBox.addEventListener('change', () => {
+    toggleCheckBox(taskEl, taskDescription, toDoList);
   });
-};
 
-const displayToDoList = () => {
-  toDoList.forEach((task) => {
-    appendTaskEl(task);
-    handleTaskInputsClick(task.id);
+  taskDeleteIcon.addEventListener('click', () => {
+    removeTaskEl(taskEl, tasksListEl, toDoList, displayToDoList);
   });
-};
-
-const handleClearBtnClick = () => {
-  clearListBtn.addEventListener('click', () => {
-    const allTaskEl = tasksListEl.querySelectorAll('article');
 
-    allTaskEl.forEach((taskEl) => {
-      const taskCheckBox = taskEl.querySelector('input');
-      if (!taskCheckBox.checked) return;
+  taskEditIcon.addEventListener('click', () => {
+    showEditField(taskEl, taskEditInput, taskCheckBox, taskDescription, toDoList);
+  });
+};
 
-      toDoList = toDoList.filter((task) => task.id !== parseInt(taskEl.id, 10));
-      localStorage.setItem('todo', JSON.stringify(toDoList));
-      taskEl.parentElement.remove();
-    });
+const displayToDoList = () => {
+  toDoList = JSON.parse(localStorage.getItem('todo')) || [];
+  toDoList.forEach((task) => {
+    appendTaskEl(task, tasksListEl);
+    handleTaskBtnsClick(task.index, displayToDoList);
+    addDraggableListener(tasksListEl.lastElementChild, toDoList, displayToDoList);
   });
 };
 
-const handleFormSubmit = (form) => {
+const handleFormSubmit = () => {
   form.addEventListener('submit', (event) => {
     event.preventDefault();
 
-    const task = {
+    const taskObj = {
       description: form['task-description'].value,
       isCompleted: false,
-      id: toDoList.length === 0 ? 1 : toDoList.at(-1).id + 1,
+      index: toDoList.length === 0 ? 1 : toDoList.at(-1).index + 1,
     };
```

Comment:
[Optional] In order to avoid letting the user save an empty string, I'll recommend that you trim the value of the description and save it when it is not empty.